scanpy/tests/test_rank_genes_groups.py

```diff
@@ -139,6 +139,20 @@ def test_results_layers():
     for name in true_scores_t_test.dtype.names:
         assert not np.allclose(true_scores_t_test[name][:7], adata.uns['rank_genes_groups']['scores'][name][:7])
 
+@pytest.mark.xfail
+def test_singlets():
+    pbmc = pbmc68k_reduced()
+    pbmc.obs['louvain'] = pbmc.obs['louvain'].cat.add_categories(['11'])
+    pbmc.obs['louvain'][0] = '11' 
+    
+    rank_genes_groups(pbmc, groupby = 'louvain')
```

Comment:
```suggestion
def test_singlets():
    pbmc = pbmc68k_reduced()
    pbmc.obs['louvain'] = pbmc.obs['louvain'].cat.add_categories(['11'])
    pbmc.obs['louvain'][0] = '11' 
    
    with pytest.raises(ValueError, match=rf"Could not calculate statistics.*{'11'}"):
        rank_genes_groups(pbmc, groupby = 'louvain')
```

Reply:
If an `xfail` test passes, I don't think it counts as a failure. This is a more specific way to check for the correct error being thrown, and will definitely fail if the error isn't thrown.